libs/agent-sm/agent-policy/src/main/java/org/opensearch/secure_sm/policy/PolicyFile.java

```diff
@@ -308,10 +316,17 @@ public String toString() {
 
     private static class PolicyInfo {
         final List<PolicyEntry> policyEntries;
+        private final Map<ProtectionDomain, PermissionCollection> pdMapping;
 
         PolicyInfo() {
-            policyEntries = new ArrayList<>();
+            policyEntries = Collections.synchronizedList(new ArrayList<PolicyEntry>());
```

Comment:
It looks like this list is only mutated upon construction. Does that seem right? Assuming so, I'd like to do a small refactoring where a mutable list is used during construction, but then an immutable copy is passed to this class via `List.copyOf` and then there's no need for any synchronization on this list.

Reply:
perfect. thanks.

Reply:
> It looks like this list is only mutated upon construction. Does that seem right?

you are right.